tests/bluetooth/audio/mocks/include/bap_unicast_server_expects.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MOCKS_BAP_UNICAST_SERVER_EXPECTS_H_
+#define MOCKS_BAP_UNICAST_SERVER_EXPECTS_H_
+
+#include <zephyr/bluetooth/audio/bap.h>
+
+#include "bap_unicast_server.h"
+#include "expects_util.h"
+
+#define expect_bt_bap_unicast_server_cb_config_called_once(_conn, _ep, _dir, _codec)               \
+do {                                                                                               \
+	const char *func_name = "bt_bap_unicast_server_cb.config";                                 \
+												   \
+	zassert_true(mock_bap_unicast_server_cb_config_fake.call_count > 0,                        \
+		     "'%s()' was not called", func_name);                                          \
+	zassert_equal(1, mock_bap_unicast_server_cb_config_fake.call_count,                        \
+		      "'%s()' was called more than once", func_name);                              \
+												   \
+	IF_NOT_EMPTY(_conn, (                                                                      \
+		     zassert_equal_ptr(_conn, mock_bap_unicast_server_cb_config_fake.arg0_val,     \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "conn");))                                       \
+												   \
+	IF_NOT_EMPTY(_ep, (                                                                        \
+		     zassert_equal_ptr(_ep, mock_bap_unicast_server_cb_config_fake.arg1_val,       \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "ep");))                                         \
+												   \
+	IF_NOT_EMPTY(_dir, (                                                                       \
+		     zassert_equal(_dir, mock_bap_unicast_server_cb_config_fake.arg2_val,          \
+				       "'%s()' was called with incorrect '%s' value",              \
+				       func_name, "_dir");))                                       \
```

Comment:
I'm having a little trouble understanding the `IF_NOT_EMPTY` macro. 

What does "empty" mean in this context, and how can a value like `_dir` be empty?

Reply:
`EMPTY` is the macro that has no replacement expression, like:
```
#define EMPTY
#define ANY
```
the `EMPTY` is defined in `include/zephyr/sys/util_macro.h`.

The `IF_NOT_EMPTY` definition evaluates to the code below:
```
#define IF_NOT_EMPTY(_a, _code) COND_CODE_1(UTIL_BOOL(IS_EMPTY(_a)), (), _code)
```
and works like this: `IS_EMPTY(_a)` evaluates to `true` or `false` depending on `_a` has replacement expression. `_a` can be variable, macro, etc. `UTIL_BOOL(_x)` evaluates to 1 or 0. Finally, `COND_CODE_1` makes the `_code` to be compiled only if `_a` has replacement expression.

The purpose of it is to have a possibility to verify only a subset of function parameters under test.

Consider, the function `foo(a, b)` is expected to be called. Parameter `a` is the one we want to verify, but we want to skip `b` verification. With the use of `IF_NOT_EMPTY` one can write a verification function that will skip the parameter verification
```
#define SKIP_THIS_ONE
....
    expect_foo_too_be_called(a, SKIP_THIS_ONE);
```

Reply:
Ah, thanks! That makes sense to me now :)